guard-lambda/src/main.rs

```diff
@@ -1,44 +1,47 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-use std::error::Error;
-
 use cfn_guard;
-use lambda_runtime::{error::HandlerError, lambda, Context};
-use log::{self, info};
+use lambda_runtime::{handler_fn, Context, Error};
+use log::{self, LevelFilter, info};
 use serde_derive::{Deserialize, Serialize};
-use simple_logger;
+use simple_logger::SimpleLogger;
 
 #[derive(Deserialize, Debug)]
 struct CustomEvent {
     #[serde(rename = "data")]
     data: String,
     #[serde(rename = "rules")]
-    rules: String,
+    rules: Vec<String>,
 }
 
 #[derive(Serialize)]
 struct CustomOutput {
-    message: String,
+    message: Vec<serde_json::Value>,
 }
 
-fn main() -> Result<(), Box<dyn Error>> {
-    simple_logger::init_with_level(log::Level::Info)?;
-    lambda!(call_cfn_guard);
+#[tokio::main]
+async fn main() -> Result<(), Error> {
 
+    SimpleLogger::new().with_level(LevelFilter::Info).init().unwrap();
+    let func = handler_fn(call_cfn_guard);
+    lambda_runtime::run(func).await?;
```

Comment:
What is the advantage of using `async-await` here?

Reply:
`lambda-runtime` v 0.3.0 expects a `Future` trait to be implemented for handler. I have basically followed the example in `aws-lambda-rust-runtime` [repo](https://github.com/awslabs/aws-lambda-rust-runtime/blob/master/lambda-runtime/examples/basic.rs).